pkg/upstream/cluster/original_dst.go

```diff
@@ -83,6 +91,10 @@ func (lb *OriginalDstLoadBalancer) ChooseHost(lbCtx types.LoadBalancerContext) t
 		dstAdd = dstAdd + ":80"
 	}
 
+	if mosnctx.Get(ctx, types.ContextKeyListenerType) == v2.INGRESS {
```

Comment:
why use the localhost addrs instend of `dstAdd` ? [istio-iptables](https://github.com/istio/cni/blob/master/tools/packaging/common/istio-iptables.sh#L28)  -u: Specify the UID of the user for which the redirection is not applied. Typically, this is the UID of the proxy container.

Reply:
there are virtualInbound and virtualOutbound listening on 15006 and 15001 respectively. both of them are original_dst.

when dealing with virtualInbound, if use `dstAdd`, then the upstream will be the pod itself, and then make call to such host will intercept into virtualInbound again, and will choose host to isself again. this will never break out until continuously opening new sockets exhaust all memory, then OOMKilled.

Reply:
@tedli [istio-iptables.sh](https://github.com/istio/cni/blob/master/tools/packaging/common/istio-iptables.sh#L28) -u `Specify the UID of the user for which the redirection is not applied. Typically, this is the UID of the proxy container`.
So I don't think iptables will be hijacking MOSN traffic. Is there another scenario?

Reply:
The `istio-iptables` changed significantly since 1.8.x (version need confirm), that implemented in golang instead of shell script.
And for outbound 15001, traffic from sidecar process with the specified uid will be bypassed. However the inbound 15006 didn't act the same.

UID special treat:
https://github.com/istio/istio/blob/e4b360c20c84a94949add00aebf2ca7d341d4bc3/tools/istio-iptables/pkg/cmd/run.go#L334